.github/workflows/render-md-mermaid.yml

```diff
@@ -0,0 +1,16 @@
+# https://github.com/nielsvaneck/render-md-mermaid#github-action--workflow-example
+name: render-md-mermaid
+on: push
+jobs:
+  render-md-mermaid:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Render images for every Mermaid diagram in all Markdown files (*.md) in the repo
+        uses: nielsvaneck/render-md-mermaid@v2
+      - name: Commit rendered png and svg files
+        uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
Are only new and updated images committed? 
If the mermaid source doesn't change there should not be any new commits or uncommitted files in the repo after running this github action. Uncommitted files, would make the publishing fail.

How does it detect, that a new version of the file needs to be created when the mermaid source changes? Or does it blindly create an image every time until the created image is not the same anymore.

Another question, GitHub is caching images referenced in Markdown files. Change the image file in the repo won't clear the cache and the old image is shown. That's why in doc/images there a files with version numbers to make rendering work.

Reply:
> Are only new and updated images committed? If the mermaid source doesn't change there should not be any new commits or uncommitted files in the repo after running this github action. Uncommitted files, would make the publishing fail.

Agreed, and these were my expectations too while cooking that action; will remove it all very soon though.

> How does it detect, that a new version of the file needs to be created when the mermaid source changes? Or does it blindly create an image every time until the created image is not the same anymore.

Action itself was yet to be tested (in action), but the test script mentioned in this original commit message dirtied the image files only when their mermaid source code changed; AFAIK that was.

> Another question, GitHub is caching images referenced in Markdown files. Change the image file in the repo won't clear the cache and the old image is shown. That's why in doc/images there a files with version numbers to make rendering work.

Good point, which mainly leads to me pushing another approach next.